eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -155,23 +159,23 @@ public InstanceInfo(
             @JsonProperty("lastDirtyTimestamp") Long lastDirtyTimestamp,
             @JsonProperty("actionType") ActionType actionType,
             @JsonProperty("asgName") String asgName) {
-        this.appName = appName;
-        this.appGroupName = appGroupName;
-        this.ipAddr = ipAddr;
-        this.sid = sid;
+        this.appName = appName!=null ? appName.intern():null;
+        this.appGroupName = appGroupName!=null? appGroupName.intern():null;
+        this.ipAddr = ipAddr!=null?ipAddr.intern():null;
```

Comment:
Is the savings mostly because this data is repeated for many instances of the sample application? Why would you have many InstanceInfo objects with the same ip address? With recent jvms the default string table size is around 60k and I would expect interning the ips would quickly lead to collisions causing the performance to go down.